hooks/tk-multi-publish2/basic/nuke_publish_reviewsubmission.py

```diff
@@ -0,0 +1,333 @@
+        item_filters property will be presented to this method.
+
+        A publish task will be generated for each item accepted here. Returns a
+        dictionary with the following booleans:
+
+            - accepted: Indicates if the plugin is interested in this value at
+                all. Required.
+            - enabled: If True, the plugin will be enabled in the UI, otherwise
+                it will be disabled. Optional, True by default.
+            - visible: If True, the plugin will be visible in the UI, otherwise
+                it will be hidden. Optional, True by default.
+            - checked: If True, the plugin will be checked in the UI, otherwise
+                it will be unchecked. Optional, True by default.
+
+        :param settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+
+        :returns: dictionary with boolean keys accepted, required and enabled
+        """
+
+        accepted = True
+        # if a publish template is configured, disable context change. This
+        # is a temporary measure until the publisher handles context switching
+        # natively.
+        if settings.get("Publish Template").value:
+            item.context_change_allowed = False
+        review_submission_app = self.parent.engine.apps.get("tk-multi-reviewsubmission")
+        if review_submission_app is None:
+            accepted = False
+            self.logger.warning(
+                "Review submission app is not available. skipping item: %s" %
+                (item.properties["publish_name"],)
+            )
+        write_node_app = self.parent.engine.apps.get("tk-nuke-writenode")
+        if write_node_app is None:
+            accepted = False
+            self.logger.warning(
+                "Write Node app is not available. skipping item: %s" %
+                (item.properties["publish_name"],)
+            )
+
+        if accepted:
+            self.logger.info(
+                "Nuke review submission item '%s' is accepted by the plugin." %
+                (item.properties["publish_name"],)
+            )
+        return {
+            "accepted": accepted,
+            "checked": True
+        }
+
+    def validate(self, settings, item):
+        """
+        Validates the given item to check that it is ok to publish. Returns a
+        boolean to indicate validity.
+
+        :param settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+        :returns: True if item is valid, False otherwise.
+        """
+
+        review_submission_app = self.parent.engine.apps.get("tk-multi-reviewsubmission")
+        if review_submission_app is None:
```

Comment:
We already checked above in `accept()`. I think we can remove this check.

Reply:
removed